intake/cli/server/server.py

```diff
@@ -86,17 +87,21 @@ class ServerInfoHandler(tornado.web.RequestHandler):
     def initialize(self, cache, catalog, auth):
         self.cache = cache
         self.catalog = catalog
+        self.searches = {}  # Map search queries to cached sources.
         self.auth = auth
 
     def get(self):
         head = self.request.headers
         page_size = self.get_argument('page_size', None)
         page_offset = self.get_argument('page_offset', 0)
+        query_list = json.loads(self.get_argument('query', '[]'))
         if self.auth.allow_connect(head):
             if 'source_id' in head:
                 cat = self.cache.get(head['source_id'])
+                query_key = (head['source_id'],)
```

Comment:
Don't we just operate on `cat`? (or is this only for making the cat have a nice name)

Reply:
I'm not sure I understand the question. This is for creating a unique (internal) key to cache search results to be generated below. The key needs to capture both the search query itself and some identification of which (sub)catalog the search was performed on. This line does that second part.